### PR TITLE
tidb: upgrade username length limit to 32

### PR DESCRIPTION
**Pull request by tiancaiamao**
There're some feedback from the user that 16 byte length limit of username is too short, 
and MySQL 5.7 already adjust the limit to 32.
https://dev.mysql.com/doc/refman/5.7/en/user-names.html

This PR change the limit to 32.

@shenli

### CODE REVIEW

File: bootstrap.go [hunk 1]
```diff
@@ -552,6 +557,10 @@ func upgradeToVer16(s Session) {
 	doReentrantDDL(s, "ALTER TABLE mysql.stats_histograms ADD COLUMN `cm_sketch` blob", infoschema.ErrColumnExists)
 }
 
+func upgradeToVer17(s Session) {
+	doReentrantDDL(s, "ALTER TABLE mysql.user MODIFY User CHAR(32)")
```
Comment (shenli) [inline bootstrap.go]: Should we also modify the create table statement of mysql.user?

Comment (tiancaiamao) [reply bootstrap.go]: Done

**Comment by zz-jason** [comment]
LGTM

**Pull request merged**